k8s/serviceAccount-orderer.yaml

```diff
@@ -0,0 +1,31 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: substra-delete-hook
+  namespace: orderer
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: substra-delete-hook
+  namespace: orderer
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs:
+    - delete
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: substra-delete-hook
+  namespace: orderer
+subjects:
+- kind: ServiceAccount
+  name: substra-delete-hook
+  namespace: orderer
+roleRef:
+  kind: Role
+  name: substra-delete-hook
+  apiGroup: rbac.authorization.k8s.io
```

Comment:
What about putting those in the chart, but with a boolean flag to active them ?

**values.yaml**

```
postHook:
  serviceAccount: substra-delete-hook # must be set to a SA which has delete permissions on secrets
  # createServiceAccount: substra-delete-hook # alternatively, uncomment this line to auto-create the SA
```

**skaffold.yaml**

```
postHook:
  createServiceAccount: substra-delete-hook
```

Reply:
The serviceAccount would be part of the release and would get deleted before the post-delete hook starts.